plugins/QnA/class.qna.plugin.php

```diff
@@ -1113,10 +1113,21 @@ public function postController_afterForms_handler($sender) {
      * @param PostController $sender Sending controller instance.
      */
     public function postController_question_create($sender, $categoryUrlCode = '') {
-            // Create & call PostController->discussion()
-            $sender->View = PATH_PLUGINS . '/QnA/views/post.php';
-            $sender->setData('Type', 'Question');
-            $sender->discussion($categoryUrlCode);
+        $categoryModel = new CategoryModel();
+        if ($categoryUrlCode != '') {
+            $category = (array)$categoryModel->getByCode($categoryUrlCode);
+            $category = $categoryModel::permissionCategory($category);
+            $isAllowedTypes = isset($category['AllowedDiscussionTypes']);
+            $isAllowedQuestion = in_array('Question', $category['allowedDiscussionType']);
+        }
+
+        if ($category && !$isAllowedQuestion && $isAllowedTypes) {
+            $sender->Form->addError(t('You are not allowed to post a question in this category.'));
```

Comment:
Shouldn't we just have to check if AllowedDiscussionTypes has type 'Question' in it?  $isAllowedTypes I don't believe is necessary